tools/sst_dump_tool.cc

```diff
@@ -404,6 +405,17 @@ Status SstFileDumper::ReadSequential(bool print_kv, uint64_t read_num,
                 blob_index.DebugString(output_hex_).c_str());
       }
     }
+
+    if (last_key.size() != 0 &&
+        internal_comparator_.Compare(key, Slice(last_key)) <= 0) {
```

Comment:
Couldn't this cause unnecessary error if the database comparator is unavailable to sst_dump?